src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -278,7 +278,8 @@ public void databaseClosing() {
     protected void close() {
         // shutdown ScheduledExecutorService. Without shutdown, cause thread leak
         if (remoteRequestExecutor != null && !remoteRequestExecutor.isShutdown()) {
-            Utils.shutdownAndAwaitTermination(remoteRequestExecutor);
+            // Note: Time to wait is set 60 sec because RemoteRequest's socket timeout is set 60 seconds.
+            Utils.shutdownAndAwaitTermination(remoteRequestExecutor, 60);
         }
```

Comment:
No. The remote request is not aborted immediately. RemoteRequest is blocking till timeout which is 60 seconds. 

It seems HttpClient does not have cancel (abort) feature.
https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/